src/game/Pet.cpp

```diff
@@ -164,6 +247,16 @@ bool Pet::LoadPetFromDB(Player* owner, uint32 petentry, uint32 petnumber, bool c
         }
     }
 
+    if (!ignoreDead)
+    {
+        uint32 pet_health = fields[13].GetUInt32();
```

Comment:
you can move this outside, so you don't have to get it again fields[13].GetUInt32();